ec2/ebs-encryption-defaults/src/handlers.ts

```
@@ -56,6 +56,8 @@ class Resource extends BaseResource<ResourceModel> {
                         await ec2client.modifyEbsDefaultKmsKeyId({ KmsKeyId: model.defaultEbsEncryptionKeyId}).promise();
                     }
                 }
+            } else {
+                throw new Error('no aws session found - did you forget to register the execution role?');
```

For consistency we should throw the internalfailure error

---

yes, done. the real gain would be to apply and refactor the handlers into using something like https://github.com/org-formation/aws-resource-providers/blob/master/s3/public-access-block/src/common.ts